frontend/test/metabase/scenarios/visualizations/funnel.cy.spec.js

```diff
@@ -0,0 +1,76 @@
+import { restore, visitQuestionAdhoc, sidebar } from "__support__/e2e/helpers";
+
+import { SAMPLE_DB_ID } from "__support__/e2e/cypress_data";
+import { SAMPLE_DATABASE } from "__support__/e2e/cypress_sample_database";
+
+const { PEOPLE_ID, PEOPLE } = SAMPLE_DATABASE;
+
+describe("scenarios > visualizations > funnel chart", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsNormalUser();
+
+    visitQuestionAdhoc({
+      dataset_query: {
+        type: "query",
+        query: {
+          "source-table": PEOPLE_ID,
+          aggregation: [["count"]],
+          breakout: [["field", PEOPLE.SOURCE]],
+        },
+        database: SAMPLE_DB_ID,
+      },
+      display: "funnel",
+    });
+    cy.findByText("Settings").click();
+    sidebar().findByText("Data").click();
+  });
+
+  it("should show the list of rows", () => {
+    cy.findAllByTestId(/draggable-item/).should("have.length", 5);
```

Comment:
In our codebase we tend to write too short E2E tests, however, the proper use would be to have longer user flows. Unit and integration tests are there to check a single aspect of a UI element or logic but doing so with E2E tests is costly since for every test it will restore a DB snapshot, perform the same login steps, etc. I would suggest to combine these single thing checks, mabye divide it with comments describing what you are verifying

Reply:
Sounds good! Will do